promql/engine.go

```diff
@@ -2049,10 +2113,18 @@ func (ev *evaluator) eval(ctx context.Context, expr parser.Expr) (parser.Value,
 		return mat, ws
 
 	case *parser.MatrixSelector:
+
+		vs := e.VectorSelector.(*parser.VectorSelector)
+		offset := durationMilliseconds(vs.Offset)
+		var mint, maxt int64
 		if ev.startTimestamp != ev.endTimestamp {
-			panic(errors.New("cannot do range evaluation of matrix selector"))
```

Comment:
Shouldn't this error still appear for the cases in which it did previously? E.g. if someone tries to run `foo[5m]` as a range query? Or was this removed since that error case is already checked in the parser (where it returns "invalid expression type "range vector" for range query, must be Scalar or instant Vector")?